### PR TITLE
Changes from Fall Game Jam 2021

### PR DESCRIPTION
**Pull request by EMattfolk**
This should be everything, but I may have missed some things.

### CODE REVIEW

File: sylt-compiler/src/typechecker.rs [hunk 1]
```diff
@@ -574,6 +574,7 @@ impl TypeChecker {
                         );
                         self.stack.push(var);
                     }
+                    self.check_constraints(span, ctx, to_match)?;
```
Comment (FredTheDino) [inline sylt-compiler/src/typechecker.rs:577]: We should add tests for this!!!

File: sylt-compiler/src/preamble.lua [hunk 2]
```diff
@@ -282,13 +282,13 @@ end
 
 __BLOB_META = { _type = "blob" }
 __BLOB_META.__eq = function(a, b)
-    for k, _ in pairs(a) do
-        if not b[k] then
+    for k, v in pairs(a) do
+        if b[k] == nil or v ~= b[k] then
```
Comment (FredTheDino) [inline sylt-compiler/src/preamble.lua:286]: We should test this!!! And we shouldn't need to test for null... Righ?

**Review by FredTheDino — COMMENTED** [review]
Looks good, but we need to test more!

File: sylt-compiler/src/typechecker.rs (hunk 1 above)
Comment (EMattfolk) [reply sylt-compiler/src/typechecker.rs:577]: I can't remember what it fixed :nog:

**Review by FredTheDino — APPROVED** [review]
LGTM!
